### ISSUE TITLE
Revisions beyond 25 are not getting stored in ios 

### ISSUE THREAD

**Issue by sridevi-15** [couchbase/couchbase-lite-ios#2053]
CBL - 2.0- DB 22 -> ios build 574
client -> ios
sg -> 2.0.0- 773

Steps to reproduce :   
 1.  Create docs in CBL and update docs 4 times.
2. set up sync-gateway config with revs limit 35
 3.  Start a continuous Replicator with push_pull to have SG load all the docs. Verify if the no. of docs are same in both SG and CBL.
4. Create a conflict in sync-gateway. 
 5.. Update docs in CBL with number of times more than revs_limit i.e more than 35
6. Wait until replicator is idle to make sure replication is completed.
 7.Get revision history in sync-gateway with following api for all docs : 
  http://192.168.33.11:4984/db/<doc_name>?conflicts=true&revs=true
8. Verify revision in history shows all 35 revisions as the revision limit is 35.

Result : 
it shows only 23 revisions.
Note : if I do not create conflict at step 4, then it works as expected.
Same steps including step 4, it works fine in Android

Test code : https://github.com/couchbaselabs/mobile-testkit/blob/feature/cbl20-ver1/testsuites/CBLTester/CBL_Functional_tests/test_no_conflicts_cbl.py#L202 

Logs : https://gist.github.com/sridevi-15/adb25160a397c632ada2269623d28f70
Errors are neither in CBL logs nor in sync-gateway logs

**Comment by pasin** [comment]
@sridevi-15 Can you also enable Database log? Please update the full log both client and SG.

**Comment by pasin** [comment]
Also the python test code [here](https://github.com/couchbaselabs/mobile-testkit/blob/feature/cbl20-ver1/testsuites/CBLTester/CBL_Functional_tests/test_no_conflicts_cbl.py#L198) set the rev_limits parameters to 25. There is some mismatching.

**Comment by sridevi-15** [comment]
Hi Pasin, I tested with parameter 35 when I run locally , it stores only 23 revisions. When I tried with 25, it gave 22 revisions from history

**Comment by pasin** [comment]
Can you also share the SG config? For me, SG always has only 20 revs in the history. Probably I don't configure the SG properly.

**Comment by pasin** [comment]
I have noticed that android and .NET don't have https://github.com/couchbase/couchbase-lite-ios/commit/8b67103c709dbde40306a1b9b2d753e81880b9bd change in for beta.

I think this is a side effect of that change. What I guessed is that the pull replicator pulled a revision from SG and the pull revision has a conflict as the same revision was pruned. The new way of saving resolved revision could cause the branch that contains the recent doc modification from the test code.

After talking to @djpongh, I should back the change out to make iOS having the same behavior as android and .NET.
